python/ray/experimental/joblib/_raybackend.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import ray
+from joblib._parallel_backends import MultiprocessingBackend
+from joblib.pool import MemmappingPool, PicklingPool
+from ray.experimental.multiprocessing.pool import Pool
+RAY_ADDRESS_ENV = "RAY_ADDRESS"
+logger = logging.getLogger(__name__)
+
+
+class RayBackend(MultiprocessingBackend):
+    """Ray backend uses ray, a system for scalable distributed computing.
+    more info are available here: https://ray.readthedocs.io
+    """
+
+    def configure(self,
+                  n_jobs=1,
+                  parallel=None,
+                  prefer=None,
+                  require=None,
+                  **memmappingpool_args):
+        # Make Ray Pool the father class of PicklingPool
+        PicklingPool.__bases__ = (Pool, )
+        MemmappingPool.__bases__ = (PicklingPool, )
+        """Use all available resources when n_jobs == -1. Must set RAY_ADDRESS
+        variable in the environment or run ray.init(address=..) to run on
+        multiple nodes.
+        """
+        if n_jobs == -1:
+            if not ray.is_initialized():
+                import os
+                if RAY_ADDRESS_ENV in os.environ:
+                    ray_address = os.environ[RAY_ADDRESS_ENV]
+                    logger.info(
+                        "Connecting to ray cluster at address='{}'".format(
+                            ray_address))
+                    ray.init(address=ray_address)
+                else:
+                    logger.info("Starting local ray cluster")
+                    ray.init(num_cpus=None)
```

Comment:
is there a reason not to use all the local available CPU resources?

Reply:
I am not sure I understand. Doesn't num_cpus=None do that?

Reply:
```suggestion
                    ray.init()
```